ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -23,11 +23,45 @@
 from ros2_generate_interface_docs import srv_utils
 from ros2_generate_interface_docs import utils
 
+from rosidl_runtime_py import get_action_interfaces
 from rosidl_runtime_py import get_message_interfaces
 from rosidl_runtime_py import get_service_interfaces
 
 
-def generate_interfaces(interfaces, html_dir, template, interface_type):
+def generate_interfaces_index(messages, services, actions, html_dir, timestamp):
+    """
+    Generate index for packages.
+
+    :param messages: Dictionay with the message package as a key and a list message names as value
+    :param type: dict
+    :param services: Dictionay with the service package as a key and a list sercice names as value
+    :param type: dict
+    :param actions: Dictionay with the action package as a key and a list action names as value
+    :param type: dict
+    :param html_dir: path to the directory to save the index
+    :type html_dir: str
+    :param timestamp: timestamp to include in the index site
+    :param timestamp: time.struct_time
+    """
+    interface_packages = set(list(messages.keys()) + list(services.keys()) + list(actions.keys()))
```

Comment:
It looks like the union of dictionary keys can be just as simple as a set union.
```
interface_packages = message.keys() | service.keys() | actions.keys()
```